### PR TITLE
Fix SVD instability bug

### PR DESCRIPTION
**Pull request by YuhanLiin**
Fixes the SVD bug in #983 and introduces a new test covering it.

### CODE REVIEW

File: tests/linalg/svd.rs [hunk 1]
```diff
@@ -460,3 +460,27 @@ fn svd_sorted() {
         epsilon = 1.0e-5
     );
 }
+
+#[test]
+// Exercises bug reported in issue #983 of nalgebra
+fn svd_consistent() {
```
Comment (Andlon) [inline tests/linalg/svd.rs:466]: maybe rename to `svd_regression_issue_983` for clarity? Adding a link to the issue might also be nice :)

**Review by Andlon — COMMENTED** [review]
Thanks for looking into this! At first glance this looks great, added a few minor comments.

I think your change of epsilon might fix the example in #1072 as well. Would you be willing to add a regression test for the example in that issue as well? That would be great!

**Comment by Andlon** [comment]
@sebcrozet: would be great if you could approve the CI :-)

**Comment by Andlon** [comment]
@sebcrozet: this looks good to me, but I'm not intimately familiar with the SVD algorithm, so would be great if you could take a closer look as well when you have time.

File: src/linalg/svd.rs [hunk 2]
```diff
@@ -888,13 +888,13 @@ fn compute_2x2_uptrig_svd<T: RealField>(
...
-            let (csu, sgn_u) = GivensRotation::new(cu, su);
+        let cu = (m11.scale(csv.c()) + m12 * csv.s()) / v1.clone();
+        let su = (m22 * csv.s()) / v1.clone();
+        let (csu, sgn_u) = GivensRotation::new(cu, su);
+        v1 *= sgn_u.clone();
+        v2 *= sgn_u;
 
-            v1 *= sgn_u.clone();
-            v2 *= sgn_u;
+        if compute_u {
```
Comment (sebcrozet) [inline src/linalg/svd.rs:897]: Why was this change needed?

**Comment by sebcrozet** [comment]
Looks good to me, thanks!
